Update Get-WebCertificate function to work in new .NET Versions

Somberous opened the pull request:
Due to the changes to [Net.HttpWebRequest], it no longer returns certificate information with the request.

To fix this, we must replace the usage of [Net.HttpWebRequest] with a method using System.Net.Sockets.TcpClient and System.Net.Security.SslStream. This approach allows us to directly establish a TCP connection to the target server and retrieve its SSL/TLS certificate using SslStream.RemoteCertificate.

This solution works in PowerShell 7, and Windows Powershell 5.1. Additionally, these changes ensure cross-platform compatibility, as it uses .NET Core's supported classes.

To learn more about the deprecation of this class please checkout these links:

    https://learn.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest?view=net-8.0
    https://learn.microsoft.com/en-us/dotnet/core/compatibility/networking/6.0/webrequest-deprecated

Somberous committed:
Update Get-WebCertificate function to work in new .NET Versionss

Due to the changes to [Net.HttpWebRequest], it no longer returns certificate information with the request.

To fix this, we must replace the usage of [Net.HttpWebRequest] with a method using System.Net.Sockets.TcpClient and System.Net.Security.SslStream. This approach allows us to directly establish a TCP connection to the target server and retrieve its SSL/TLS certificate using SslStream.RemoteCertificate.

This solution works in PowerShell 7, and Windows Powershell 5.1. Additionally, these changes ensure cross-platform compatibility, as it uses .NET Core's supported classes.

To learn more about the deprecation of this class please checkout these links:

    https://learn.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest?view=net-8.0
    https://learn.microsoft.com/en-us/dotnet/core/compatibility/networking/6.0/webrequest-deprecated

masters274 reviewed (changes requested):
In order for the updates to be pushed into the PowerShell Gallery upon merging the PR, you'll need to increment the module manifest, and update the notes explaining the changes made for the version. Please update, following the template of the existing notes

Somberous committed:
Update Network.psd1

Updated module manifest to reflect the changes made within the PR for the Get-WebCertificate function.

Somberous commented:
Requested changes have been added to https://github.com/Somberous/Network/tree/patch-1.
